build.rs

```diff
@@ -351,7 +347,105 @@ fn build_msvc(target_triple: &[&str], disable_opt: bool, num_jobs: &str,
     let mut test_args = args.clone();
     test_args.push(String::from("crypto/libring-test.Windows.vcxproj"));
     run_command_with_args(&msbuild, &test_args);
-    Ok(())
+    println!("cargo:rustc-link-search=native={}", lib_path.to_str().unwrap());
+    println!("cargo:rustc-link-lib=static={}-core", LIB_NAME);
+
+    // XXX: Ideally, this would only happen for `cargo test`, but we don't know
+    // how to do that yet.
+    println!("cargo:rustc-link-lib=static={}-test", LIB_NAME);
+}
+
+fn build_unix(target_info: &TargetInfo, out_dir: PathBuf) {
+    let mut lib_target = out_dir.clone();
+    lib_target.push("libring-core.a");
+    let lib_target = lib_target.as_path();
+
+    let mut test_target = out_dir.clone();
+    test_target.push("libring-test.a");
+    let test_target = test_target.as_path();
+
+    let lib_header_change = RING_HEADERS.iter()
+        .map(Path::new)
+        .any(|p| need_run(&p, lib_target));
+    let test_header_change = RING_TEST_HEADERS.iter()
+        .map(Path::new)
+        .any(|p| need_run(&p, test_target)) ||
+                             lib_header_change;
```

Comment:
I suggest that we just check if any of `find crypto include mk -name "*.h" -o -name "*.inl" -o -name "*.mk"` changed, and if so, rebuild all the libraries. Especially since the test lib will go away sometime soon.